src/gtk/textctrl.cpp

```diff
@@ -1017,6 +1023,55 @@ void wxTextCtrl::GTKSetJustification()
     }
 }
 
+#if wxUSE_SPELLCHECK
+
+bool wxTextCtrl::EnableSpellCheck(const bool enable)
+{
+    wxCHECK_MSG( IsMultiLine(), false,
+                "Unable to enable spell check on control "
+                "which does not have wxTE_MULTILINE style" );
+
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+    wxCHECK_MSG( textview, false, wxS("wxTextCtrl is not a GtkTextView" ));
+
+    wxString lang = wxLocale::GetLanguageCanonicalName(wxLANGUAGE_DEFAULT);
+
+    GtkSpellChecker *spell = gtk_spell_checker_get_from_text_view(textview);
+
+    if ( enable )
+    {
+        if ( !spell )
+        {
+            spell = gtk_spell_checker_new();
+            gtk_spell_checker_attach(spell, textview);
+        }
+
+        if ( !gtk_spell_checker_set_language(spell, NULL, NULL) )
+            return false;
+    }
+    else
+    {
+        if ( spell )
+            gtk_spell_checker_detach(spell);
+    }
+
+   return true;
+}
+
+bool wxTextCtrl::IsSpellCheckEnabled() const
+{
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+
+    if ( !IsMultiLine() || textview == NULL)
```

Comment:
Extremely minor, but
```suggestion
    if ( !IsMultiLine() || textview == NULL )
```

Reply:
> Thanks again, but I think changes to `in{clude,terface}/wx/textctrl.h` got accidentally left out, could you please add them?

Sorry, they are added now.

> We also should ensure that the package containing `gtkspell.h` is installed during the CI builds to check compilation of this code.

Is this not checked by PKG_CHECK_MODULES(GTKSPELL, [gtkspell3-3.0], ...? I will make the changes to configure.in that you mentioned above - but is there another step needed?

> Finally, it looks like it should be simple to implement this for macOS too as we already have `MacCheckSpelling()` there.

Yes, it looks almost trivial except for the IsSpellCheckEnabled() part. This (probably) requires a change to cocoa/textctrl.mm file - again quite simple, but I have no means of testing other than by the CI tools. Therefore, I would prefer to do this in a separate branch once I have finished the wxMSW and wxGTK stuff.

Reply:
> Is this not checked by PKG_CHECK_MODULES(GTKSPELL, [gtkspell3-3.0], ...? I will make the changes to configure.in that you mentioned above - but is there another step needed?

I was thinking of installing the required packages in `.github/workflows/ci.yml` as currently they're not available in the CI builds, as can be seen from
```
configure: WARNING: gtkspell3-3.0 not found, spellchecking won't be available
```
message in the build logs.

As for Mac part, I think I can do it, so please don't worry about this.

TIA!